submit_for_testing.py

```diff
@@ -120,15 +120,16 @@ def _submit_to_squad(lava_job, lava_url_base, qa_server_api, qa_server_base, qa_
             ).netloc,  # qa-reports backends are named as lava instances
         }
         logger.info("Submit to: %s" % qa_server_api)
-        results = requests.post(qa_server_api, data=data, headers=headers, timeout=31)
-        if results.status_code < 300:
-            logger.info(
-                "%s/testjob/%s %s"
-                % (qa_server_base, results.text, get_job_name(lava_job))
+        try:
+            results = requests.post(
```

Comment:
This will not throw an exception, even if it is an error. You need to call `raise_for_status` on the response.

```
results = ...
results.raise_for_status()
```

See here: https://2.python-requests.org/en/latest/user/quickstart/#response-status-codes